### DISCUSSION TITLE
_check_and_init_precision: PrecisionPlugin is enabled even _precision_plugin_flag is None

### DISCUSSION THREAD

**Discussion by wohenniubi** [Lightning-AI/pytorch-lightning#13172] [General]
In lightning 1.6.x, for the code in accelerator_connector.py#L645 ~ accelerator_connector.py#L658, 
https://github.com/PyTorchLightning/pytorch-lightning/blob/dd475183227644a8d22dca3deb18c99fb0a9b2c4/pytorch_lightning/trainer/connectors/accelerator_connector.py#L645

It's better to enable them only when **self._precision_plugin_flag** is not None. E.g.
 ```
   if self._precision_plugin_flag != None:
       if isinstance(self.accelerator, IPUAccelerator):
            return IPUPrecisionPlugin(self._precision_flag)  # type: ignore
        if isinstance(self.accelerator, HPUAccelerator):
            return HPUPrecisionPlugin(self._precision_flag)  # type: ignore
        if isinstance(self.accelerator, TPUAccelerator):
            if self._precision_flag == 32:
                return TPUPrecisionPlugin()
            elif self._precision_flag in (16, "bf16"):
                if self._precision_flag == 16:
                    rank_zero_warn(
                        "You passed `Trainer(accelerator='tpu', precision=16)` but AMP"
                        " is not supported with TPUs. Using `precision='bf16'` instead."
                    )
                return TPUBf16PrecisionPlugin()
```

**Comment by akihironitta** [comment]
Hi @wohenniubi, can I ask what makes you think so? Are you seeing any misbehaviour?

---

If a user gives a `PrecisionPlugin` (`self._precision_plugin_flag`), the accelerator connector checks if the provided plugin is a valid one in:
https://github.com/PyTorchLightning/pytorch-lightning/blob/dd475183227644a8d22dca3deb18c99fb0a9b2c4/pytorch_lightning/trainer/connectors/accelerator_connector.py#L283-L314
and in this method:
https://github.com/PyTorchLightning/pytorch-lightning/blob/dd475183227644a8d22dca3deb18c99fb0a9b2c4/pytorch_lightning/trainer/connectors/accelerator_connector.py#L702

So, if the provided `PrecisionPlugin` is valid, the following line will return the plugin as is,
https://github.com/PyTorchLightning/pytorch-lightning/blob/dd475183227644a8d22dca3deb18c99fb0a9b2c4/pytorch_lightning/trainer/connectors/accelerator_connector.py#L642-L643

but if not, it'll be guaranteed that `self._precision_plugin_flag` is `None` there, and thus we don't need to check it.
